iset.mm

```diff
@@ -34948,6 +34948,21 @@ than an implication (but the two are equivalent by ~ bm1.3ii ).
       ONFOAHOMBIJKL $.
   $}
 
+  ${
+    $d ph x y z $.
+    undifexmid.1 $e |- ( x C_ y <-> ( x u. ( y \ x ) ) = y ) $.
+    $( If the union of complementary parts always produces the whole, excluded
+       middle follows.  (Contributed by Jim Kingdon, 16-Jun-2022.) $)
+    undifexmid $p |- ( ph \/ -. ph ) $=
```

Comment:
The statement doesn't match the comment. Are you using the biconditional in the proof, or only the right implication ?

Reply:
Seems to me that the hypothesis should exactly be http://us.metamath.org/mpeuni/undif.html (with class changed to set). If so, it is the comment which needs adjusting (the word "always" in particular but maybe other bits).

Reply:
How does

```
Union of complementary parts producing the whole and excluded middle.
Although special cases such as ~ undifss and ~ undifdcss are provable,
the full statement implies excluded middle as shown here.
```

look?

Reply:
I understand your argument.
Note that my proposed hyothesis can be read: "every subset is complemented", which also has its charm.
As you prefer.
And after that, you can merge.

Reply:
I'll merge this for now. If we find we miss the version phrased in terms of `y C_ ( x u. ( y \ x ) )` we can always revisit (maybe by then we'll have `EXMID` if that comes to pass).